File: typescript/02-streaming.ts
/**
 * Streaming chat (SSE). Do not buffer the full response.
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 02-streaming.ts
 */
import { Tonia } from "@tonia/sdk";

const client = new Tonia({ apiKey: process.env.TONIA_API_KEY });

const { data } = await client.models.list();
if (!data[0]) {
  throw new Error("empty allowlist — do not guess a model id");
}

for await (const event of client.chat.completions.stream({
  model: data[0].id,
  messages: [{ role: "user", content: "Compte jusqu’à 5." }],
})) {
  if (event.data === "[DONE]") break;
  if (event.json) {
    process.stdout.write(JSON.stringify(event.json) + "\n");
  }
}

File: python/02_streaming.py
"""Streaming chat (SSE). Do not buffer the full response.

    export TONIA_API_KEY=tonia_sk_…
    python 02_streaming.py
"""

from __future__ import annotations

import json
import os

from tonia import Tonia

with Tonia(api_key=os.environ["TONIA_API_KEY"]) as client:
    listed = client.models.list()
    ids = [model["id"] for model in listed["data"]]
    if not ids:
        raise SystemExit("empty allowlist — do not guess a model id")
    for event in client.chat.completions.stream(
        model=ids[0],
        messages=[{"role": "user", "content": "Compte jusqu’à 5."}],
    ):
        if event.data == "[DONE]":
            break
        if event.json is not None:
            print(json.dumps(event.json, ensure_ascii=False))
